tools/crm_simulate.c

```diff
@@ -73,6 +76,274 @@ extern gboolean bringing_nodes_online;
 	}					\
     } while(0)
 
+#define INDENT "                                   "
+
+static gboolean
+in_place_cb(const gchar *option_name, const gchar *optarg, gpointer data, GError **error) {
+    options.store = TRUE;
+    options.process = TRUE;
+    options.simulate = TRUE;
+    return TRUE;
+}
+
+static gboolean
+live_check_cb(const gchar *option_name, const gchar *optarg, gpointer data, GError **error) {
+    if (options.xml_file) {
+        free(options.xml_file);
+    }
+
+    options.xml_file = NULL;
+    return TRUE;
+}
+
+static gboolean
+node_down_cb(const gchar *option_name, const gchar *optarg, gpointer data, GError **error) {
+    options.modified++;
+    options.node_down = g_list_append(options.node_down, (gchar *) optarg);
```

Comment:
I'm guessing your issue comes from not strdup'ing optarg

Reply:
Yep, that fixes it.  What's funny is I went through a round of strdup'ing optarg for the various *_file variables but didn't think to go back and do it for these.